librocco_receipt_printer/print_job.py

```diff
@@ -14,5 +23,52 @@ class PrintJob(Document):
     TODO: Make this more elaborate, this is just a placeholder
     """
 
+    def __str__(self):
+        str = f"<PrintJob for '{self.printer_id}'\n"
+        str += f"  Items:\n"
+        for entry in self.items:
+            str += f"    isbn = '{entry['isbn']}'"
+            str += f" title = '{entry['title']}'"
+            str += f" quantity = {entry['quantity']}"
+            str += f" price = {entry['price']}\n"
+        str += f"\n  Total: {self.total}\n"
+        return str
+
+    def processing(self, db):
+        """Set the status of the print job to 'PROCESSING' and store it in the database."""
+        self.status = "PROCESSING"
+        self.store(db)
+
+    def done(self, db):
+        """Set the status of the print job to 'DONE' and store it in the database."""
+        self.status = "DONE"
+        self.store(db)
+
+    def error(self, db, error):
+        """Set the status of the print job to 'ERROR' and store it in the database."""
+        self.status = "ERROR"
+        self.error = error
+        self.store(db)
+
+    def print(self, printer):
+        """Print the print job to the printer."""
+        printer.print(self.__str__())
```

Comment:
`str(self)` is much more readable and idiomatic.

Reply:
Aaah, you mean: instead of `self.__str__()` ?
If so, it took me awhile, due to GitHub showing at least 4 lines (I'm guessing you meant just the last line)